sources/Outputs/src/Par.cpp

```diff
@@ -339,8 +341,8 @@ Par::writeGenerator(const algo::GeneratorDefinition& def, const std::string& bas
 
   //  Use the hash id in exported files to prevent use of non-ascii characters
   auto set = boost::shared_ptr<parameters::ParametersSet>(new parameters::ParametersSet(hashIdStr));
-  if(def.isUsingDiagram())
-    set->addMacroParSet(boost::shared_ptr<parameters::MacroParSet>(new parameters::MacroParSet(helper::getMacroParameterSetId(def.model, DYN::doubleIsZero(def.targetP)))));
+  // The macroParSet is associated to a macroParameterSet via the id
+  set->addMacroParSet(boost::shared_ptr<parameters::MacroParSet>(new parameters::MacroParSet(helper::getMacroParameterSetId(def.model, DYN::doubleIsZero(def.targetP)))));
```

Comment:
Why does the condition on isUsingDiagram been removed ?

Reply:
the check is already implemented (line 193) when calling the method writeGenerator, so this second condition is useless